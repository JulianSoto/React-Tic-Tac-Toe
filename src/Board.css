.board {
	display: grid;
	width: 150px;
	grid-template-columns: 50px 50px 50px;
	grid-template-rows: 50px 50px 50px;
}