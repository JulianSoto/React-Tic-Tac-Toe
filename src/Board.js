import React, {Component} from 'react';
import Square from './Square';

export default class Board extends Component {
	render(){
		return (
			<div>
				<div>
					<Square value={this.props.players[this.props.squares[0]]} onClick={this.props.squareClickHandler} index={0} />
					<Square value={this.props.players[this.props.squares[1]]} onClick={this.props.squareClickHandler} index={1} />
					<Square value={this.props.players[this.props.squares[2]]} onClick={this.props.squareClickHandler} index={2} />
				</div>
				<div>
					<Square value={this.props.players[this.props.squares[3]]} onClick={this.props.squareClickHandler} index={3} />
					<Square value={this.props.players[this.props.squares[4]]} onClick={this.props.squareClickHandler} index={4} />
					<Square value={this.props.players[this.props.squares[5]]} onClick={this.props.squareClickHandler} index={5} />
				</div>
				<div>
					<Square value={this.props.players[this.props.squares[6]]} onClick={this.props.squareClickHandler} index={6} />
					<Square value={this.props.players[this.props.squares[7]]} onClick={this.props.squareClickHandler} index={7} />
					<Square value={this.props.players[this.props.squares[8]]} onClick={this.props.squareClickHandler} index={8} />
				</div>
			</div>
		)
	}
}