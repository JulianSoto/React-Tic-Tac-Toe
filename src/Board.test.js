//board test
import React from 'react';
import Board from './Board.js';
import { shallow } from 'enzyme';

describe('Board', () => {
	it ('should match snapshot', () => {
		const wrapper = shallow(<Board squares={Array(9).fill(null)} players={['X', 'O']}/>);
		expect(wrapper).toMatchSnapshot();
	})
})