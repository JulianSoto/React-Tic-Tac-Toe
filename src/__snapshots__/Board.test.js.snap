// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Board
    players={
      Array [
        "X",
        "O",
      ]
    }
    squares={
      Array [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Square
            index={0}
          />
          <Square
            index={1}
          />
          <Square
            index={2}
          />
        </div>,
        <div>
          <Square
            index={3}
          />
          <Square
            index={4}
          />
          <Square
            index={5}
          />
        </div>,
        <div>
          <Square
            index={6}
          />
          <Square
            index={7}
          />
          <Square
            index={8}
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Square
              index={0}
            />,
            <Square
              index={1}
            />,
            <Square
              index={2}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "index": 0,
              "onClick": undefined,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "index": 1,
              "onClick": undefined,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "index": 2,
              "onClick": undefined,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Square
              index={3}
            />,
            <Square
              index={4}
            />,
            <Square
              index={5}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "index": 3,
              "onClick": undefined,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "index": 4,
              "onClick": undefined,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "index": 5,
              "onClick": undefined,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Square
              index={6}
            />,
            <Square
              index={7}
            />,
            <Square
              index={8}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "index": 6,
              "onClick": undefined,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "index": 7,
              "onClick": undefined,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "index": 8,
              "onClick": undefined,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Square
              index={0}
            />
            <Square
              index={1}
            />
            <Square
              index={2}
            />
          </div>,
          <div>
            <Square
              index={3}
            />
            <Square
              index={4}
            />
            <Square
              index={5}
            />
          </div>,
          <div>
            <Square
              index={6}
            />
            <Square
              index={7}
            />
            <Square
              index={8}
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Square
                index={0}
              />,
              <Square
                index={1}
              />,
              <Square
                index={2}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": 0,
                "onClick": undefined,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": 1,
                "onClick": undefined,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": 2,
                "onClick": undefined,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Square
                index={3}
              />,
              <Square
                index={4}
              />,
              <Square
                index={5}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": 3,
                "onClick": undefined,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": 4,
                "onClick": undefined,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": 5,
                "onClick": undefined,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Square
                index={6}
              />,
              <Square
                index={7}
              />,
              <Square
                index={8}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": 6,
                "onClick": undefined,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": 7,
                "onClick": undefined,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": 8,
                "onClick": undefined,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
